src/WooCommerce.php

```diff
@@ -249,6 +249,25 @@ public static function cron_remove_back_in_stock() {
     }
   }
 
+  /**
+   * Cron event callback to remove outdated back-in-stock product metadata.
+   */
+  public static function cron_orphan_variants_cleanup() {
+    global $wpdb;
+    $orphan_variants_ids = $wpdb->get_col(
+      "SELECT products.ID products
+      FROM {$wpdb->posts} products
+      LEFT JOIN {$wpdb->posts} wp ON wp.ID = products.post_parent
+      WHERE wp.ID IS NULL AND products.post_type = 'product_variation';"
+    );
+
+    if (!empty($orphan_variants_ids)) {
+      foreach ($orphan_variants_ids as $id) {
+        wp_delete_post($id, true);
```

Comment:
```suggestion
        wp_delete_post($id, TRUE);
```

Reply:
According to https://stackoverflow.com/questions/46874020/delete-a-product-by-id-using-php-in-woocommerce, WooCommerce uses the API of loading the product and invoking the delete method?

```php
$product = wc_get_product($id);
$product->delete(TRUE);
```

Reply:
Changed to use Woocommerce API